cmake/LLVMIRUtil.cmake

```diff
@@ -753,14 +753,26 @@ function(llvmir_attach_obj_target)
     list(APPEND IN_FULL_LLVMIR_FILES "${IN_LLVMIR_DIR}/${IN_LLVMIR_FILE}")
   endforeach()
 
-  set(FULL_OUT_LLVMIR_FILE "${WORK_DIR}/${TRGT}.o")
-  if(SHORT_NAME)
-    set(FULL_OUT_LLVMIR_FILE "${WORK_DIR}/${SHORT_NAME}.o")
-  endif()
-  get_filename_component(OUT_LLVMIR_FILE ${FULL_OUT_LLVMIR_FILE} NAME)
+  foreach(IN_LLVMIR_FILE ${IN_FULL_LLVMIR_FILES})
+    get_filename_component(OUTFILE ${IN_LLVMIR_FILE} NAME_WE)
+    set(OUT_LLVMIR_FILE "${OUTFILE}.o")
+    set(FULL_OUT_LLVMIR_FILE "${WORK_DIR}/${OUT_LLVMIR_FILE}")
 
-  list(APPEND OUT_LLVMIR_FILES ${OUT_LLVMIR_FILE})
-  list(APPEND FULL_OUT_LLVMIR_FILES ${FULL_OUT_LLVMIR_FILE})
+    add_custom_command(OUTPUT ${FULL_OUT_LLVMIR_FILE}
+        COMMAND llc
+        ARGS
+          -filetype=obj
+          -relocation-model=pic
```

Comment:
why do we need this flag here?

Reply:
I added this flag because the compiler complained in my real project that I needed to recompile with `-fPIC`.

Reply:
I'd rather not have project-specific options hardcoded like this.
This command handles other args and passes them unaltered using `${LLVMIR_ATTACH_UNPARSED_ARGUMENTS}`.

Can you please try your project with defining that flag in your `CMakeLists.txt`?